tests/auth/test_permissions.py

```diff
@@ -0,0 +1,51 @@
+from goosebit.auth import compare_permission, compare_permissions
```

Comment:
Great to have tests! They will help to ensure that this critical function does not get broken by refactoring. Also a good docu to understand what is possible.

Reading the tests I felt that `compare` likely is the wrong term - compare functions usually return "smaller", "equal", "bigger" as int. Should it rather be `check`?

Also: most tests focus on `compare_permission`. This seems a helper function only used internally to the module. So I'd rename it to `_compare_permission` (well: at the end: `_check_permission`) and change the tests to focus on the public function that is used externally to the module.